src/main/java/app/tozzi/JPASearchFunctions.java

```diff
@@ -1,71 +1,109 @@
-    public static final Function<FieldRootBuilderBean<?>, Predicate> GTE = s -> s.trim ? s.criteriaBuilder.greaterThanOrEqualTo(s.criteriaBuilder.trim(getPath(s.root, s.field).as(String.class)), s.value.toString()) : s.criteriaBuilder.greaterThanOrEqualTo(getPath(s.root, s.field), (Comparable) s.value);
-    public static final Function<FieldRootBuilderBean<?>, Predicate> LT = s -> s.trim ? s.criteriaBuilder.lessThan(s.criteriaBuilder.trim(getPath(s.root, s.field).as(String.class)), s.value.toString()) : s.criteriaBuilder.lessThan(getPath(s.root, s.field), (Comparable) s.value);
-    public static final Function<FieldRootBuilderBean<?>, Predicate> LTE = s -> s.trim ? s.criteriaBuilder.lessThanOrEqualTo(s.criteriaBuilder.trim(getPath(s.root, s.field).as(String.class)), s.value.toString()) : s.criteriaBuilder.lessThanOrEqualTo(getPath(s.root, s.field), (Comparable) s.value);
-    public static final Function<FieldRootBuilderBean<?>, Predicate> IN = s -> {
-
-        CriteriaBuilder.In<Object> in = s.criteriaBuilder.in(getPath(s.root, s.field));
-        if (s.value instanceof Collection<?> coll) {
-            for (Object obj : coll) {
-                in.value(obj);
-            }
-
-        } else {
-            in.value(s.value);
+    public static final JPAFuncWithExpressions<?, Boolean> EQ = (cb, values) -> cb.equal(values[0], values[1]);
+    public static final JPAFuncWithExpressions<String, Boolean> EQ_IGNORECASE
+        = (cb, values) -> cb.equal(cb.upper(values[0]), cb.upper(values[1]));
+    public static final JPAFuncWithExpressions<String, Boolean> STARTSWITH
+        = (cb, values) -> cb.like(values[0], cb.concat(values[1],  "%"));
+    public static final JPAFuncWithExpressions<String, Boolean> STARTSWITH_IGNORECASE
+        = (cb, values) -> cb.like(cb.upper(values[0]), cb.concat(cb.upper(values[1]), "%"));
+    public static final JPAFuncWithExpressions<String, Boolean> ENDSWITH
+        = (cb, values) -> cb.like(values[0], cb.concat("%", values[1]));
+    public static final JPAFuncWithExpressions<String, Boolean> ENDSWITH_IGNORECASE
+        = (cb, values) -> cb.like(cb.upper(values[0]), cb.concat("%", cb.upper(values[1])));
+    public static final JPAFuncWithExpressions<String, Boolean> CONTAINS
+        = (cb, values) -> cb.like(values[0], cb.concat(cb.concat("%", values[1]), "%"));
+    public static final JPAFuncWithExpressions<String, Boolean> CONTAINS_IGNORECASE
+        = (cb, values) -> cb.like(cb.upper(values[0]), cb.concat("%", cb.concat(cb.upper(values[1]), "%")));
+    public static final JPAFuncWithExpressions<?, Boolean> NOTEQ
+        = (cb, values) -> cb.notEqual(values[0], values[1]);
+    public static final JPAFuncWithExpressions<String, Boolean> NOTEQ_IGNORECASE
+        = (cb, values) -> cb.notEqual(cb.upper(values[0]), cb.upper(values[1]));
```

Comment:
Case insensitivity could be implemented with an optional third arg.

Reply:
removed and added separate "lower" operator